### PR TITLE
POC: Syntax highlighting with CodeMirror

### LINKED ISSUE #4: Syntax highlighting in the input box

### PR DESCRIPTION
**Pull request by jbadlato**
To resolve #4 

Remaining TODO:
- [x] Custom highlighting for our defined keywords

- [x] CSS to style the input box

- [x] Fix tests

### CODE REVIEW

File: src/view/InputBox.tsx [hunk 1]
```diff
@@ -19,11 +12,16 @@ interface InputBoxProps {
  * InputBox is where the user can enter their boolean expression.
  */
 export const InputBox = React.memo(function InputBox(props: InputBoxProps) {
-  const { query } = props;
-
-  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    props.onQueryChange(event.target.value);
+  const codeMirrorOptions = {
```
Comment (dpgil) [inline src/view/InputBox.tsx:15]: could pull this const out

also will python syntax highlighting work for some of our operators that don't exist in python?

File: src/view/InputBox.tsx [hunk 2]
```diff
@@ -1,14 +1,7 @@
 import React from 'react';
-
-const inputBoxStyle: React.CSSProperties = {
```
Comment (dpgil) [inline src/view/InputBox.tsx:3]: we may still want these styles on a wrapper on the codemirror object

**Comment by dpgil** [comment]
https://stackoverflow.com/questions/21572682/createtextrange-is-not-working-in-chrome/41743191

we could stub out the createRange function, or just remove that test entirely

File: src/view/InputBox.tsx (hunk 1 above)
Comment (dpgil) [reply src/view/InputBox.tsx:15]: ah i see the custom highlighting checkbox in the description

**Comment by dpgil** [comment]
looks dope

File: src/view/InputBox.tsx (hunk 1 above)
Comment (jbadlato) [reply src/view/InputBox.tsx:15]: Oh was gonna leave a comment asking - what's the React way of setting this value on initial render?  With this current code, we duplicate this default `p and q` value upstream

**Comment by jbadlato** [comment]
@dpgil  We in business :sunglasses:

File: src/view/InputBox.tsx [hunk 3]
```diff
@@ -1,14 +1,80 @@
...
+      const word: string = stream.current();
+      if (KEYWORDS.has(word)) {
+        stream.eat('word: ' + word);
+        return 'keyword';
+      }
+      stream.eat(word);
+      return null;
+    },
+  };
+});
```
Comment (jbadlato) [inline src/view/InputBox.tsx:77]: yeah yeah you probably want this bad boy in it's own file, huh? Maybe deduplicate this keywords set throughout the code base?

Comment (dpgil) [reply src/view/InputBox.tsx:77]: yeah prolly 😝

File: src/view/InputBox.tsx (hunk 1 above)
Comment (dpgil) [reply src/view/InputBox.tsx:88]: i think we could use props.query? since that's being set on initial render upstream

Comment (dpgil) [reply src/view/InputBox.tsx:88]: and you can do it on initial render with a useEffect hook and passing empty list as the second parameter

**Review by dpgil — APPROVED** [review]
:ship: :ship: :ship: ship it :ship: :ship: :ship:

**Pull request merged**